### PR TITLE
Add extension Easy Video Links

### PR DESCRIPTION
**Pull request by trey-k**
Adds new extension Easy Video Links that allows you to easily copy video URLs

### CODE REVIEW

File: Extensions/easy_video_links.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	run: function() {
+		
+		this.running = true;
+		XKit.tools.init_css("easy_video_links");
+		XKit.interface.create_control_button("xkit-easy-video-links-button", this.button_icon, "Copy Video Address", "");
```
Comment (nightpool) [inline Extensions/easy_video_links.js]: the fourth argument here should be a function that gets called when the button is clicked (equivalent to your document on click handler below)

File: Extensions/easy_video_links.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+				video_source = video_source.substring(video_source.indexOf("tumblr_"));
+				if (video_source.indexOf("/") !== -1) {
+					video_source = video_source.substring(0, video_source.indexOf("/"));
+				}
+				video_source = "https://ve.media.tumblr.com/" + video_source + ".mp4";
+			}
+			var $temp = $("<input>");
+			$("body").append($temp);
+			$temp.val(video_source).select();
+			document.execCommand("copy");
```
Comment (nightpool) [inline Extensions/easy_video_links.js]: There should be some feedback to the user that the copy took place—maybe an `XKit.notifications.add`?

Also, is there a reason we're using the old execCommand API vs the [new async clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText)? That would replace this entire dom manipulation hackery with a single function call

File: Extensions/easy_video_links.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+	run: function() {
+
```
Comment (nightpool) [inline Extensions/easy_video_links.js]: this is really nitpicky, but would you mind removing the blank lines at the beginning/end of functions?

File: Extensions/easy_video_links.js (hunk 1 above)
Comment (nightpool) [reply Extensions/easy_video_links.js]: My advice: get rid of the `init` function and instead have a `copyLink` function that serves as your click handler.

File: Extensions/easy_video_links.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+			var $temp = $("<input>");
+			$("body").append($temp);
+			$temp.val(video_source).select();
+			document.execCommand("copy");
+			$temp.remove();
+		});
+
+	},
+
+	do: function() {
```
Comment (nightpool) [inline Extensions/easy_video_links.js]: "do" is a pretty generic name. maybe `add_link_to_posts`?

**Comment by trey-k** [comment]
Alright I forgot that we didn't need to support a wide range of browsers with that copy hack but I've made the proposed changes and fixed an issue I found while testing. I could get rid of the CSS too if we merged changes to get rid of the margin and opacity changes from the old Tumblr dashboard design

**Comment by AprilSylph** [comment]
unnecessary on new dashboard; video locations can now be easily obtained with right-click -> Copy Video Location; closing

**Pull request closed without merging**